share/wake/lib/system/remote_cache_api.wake

```diff
@@ -331,16 +333,53 @@ export def rscApiFindMatchingJob (req: CacheSearchRequest) (api: RemoteCacheApi)
 #  rscApiGetStringBlob (RemoteCacheBlob "asdf" "https://...") = Pass "foo\nbar\nbat" 
 # ```
 export def rscApiGetStringBlob ((CacheSearchBlob _ uri): CacheSearchBlob): Result String Error =
-    # TODO: use "small blob" for this and also parse the schema out
-    # require False = id ==* "00000000-0000-0000-0000-000000000000"
-    # else Pass  ""
+    require scheme, path, Nil = extract `([a-zA-Z][a-zA-Z0-9+.-]*)://(.*)` uri
```

Comment:
Technically the `//` isn't required (without it the entire right side of the `:` is read as a simple payload+query+fragment;[^1] i.e. very likely something on the same machine), but since we'll likely not be dealing with any route lookups which do leave the authority off, and since the rest of the code doesn't support that case either, there's nothing strictly wrong with this as is.  Or is it an invariant you're checking when the `CacheSearchBlob` was originally created?

[^1]: You probably know this already, but for the peanut gallery, the breakdown of `https://github.com/sifive/wake/pull/1553/files#diff-UUID` winds up looking something like the below, and if we choose specific contexts to put the link in, all of them can be made optional:
    * Scheme: `https:`
    * Authority: `//github.com`
    * Path-like payload: `/sifive/wake/pull/1553/files`
    * Fragment: `#diff-UUID`

Reply:
Yeah, I intentionally slightly abuse the spec here. There isn't any technical reason that it _has_ to be this way but since every other case we have uses `://` there is nice symmetry in keeping it. The parsing is more complicated otherwise